### ISSUE TITLE
Event hookup popup dismisses after space

### ISSUE THREAD

**Issue by dpoeschl** [dotnet/roslyn#24956]
Type in C#: `Console.CancelKeyPress +=`

You'll get the "Press Tab to insert the handler" popup, but press space and the popup disappears. This is a regression in 15.6 with the editor quick info changes.

**Comment by dpoeschl** [comment]
FYI: @gundermanc -- should I make a VSO bug for this as well?

**Comment by gundermanc** [comment]
@dpoeschl by any chance have you tried this scenario in 15.7 and 15.8 bits? I've addressed some existing issues for these milestones already.

**Comment by gundermanc** [comment]
@dpoeschl I've looked into this a bit more. It looks like my changes did break this behavior, but I'm honestly puzzled as to why it was done this way in the first place instead.

The Roslyn code responsible for this behavior is changing private IDE state via reflection and using that to keep the tip open. We could fix this from our side by investigating what implementation details changed and trying to emulate it, but IMHO this falls into the bag of hacks that should not be perpetuated.

**Comment by gundermanc** [comment]
I'd suggest #24434 as the correct fix to this issue.

**Comment by dpoeschl** [comment]
Yes, agreed on all counts, except this means we're shipping 15.6 in a state where `+=<space>` no longer shows the popup, which may make users think the feature is unavailable. I was certainly surprised when it wasn't there and thought it was broken. I filed this to track the "oops, what do we do for 15.6 now" problem. 

The good news (maybe?) is that the state machine is still running even though the popup is gone, so if you hit `+=<space><tab>`, it will indeed create your handler. Not sure if this is a good workaround or even _more_ surprising.

**Comment by gundermanc** [comment]
Definitely undesirable. FWIW, I think the difference in behavior is that the old quick info was getting this _dismiss-quick-info-on-text-change_ behavior from the ShimQuickInfoController, which is a shim from 1st gen to 2nd gen quick info that isn't supposed to be running in the non-1st gen scenario.

When I redesigned quick info, I added explicit dismissal behavior to the actual quick info implementation, moving this behavior from the shims to the impl, breaking the work-around. There might be a change we could take on our side for 15.6 to restore this behavior but it's unlikely to be approved.
